python_modules/dagster/dagster/_core/definitions/definitions_class.py

```diff
@@ -884,6 +884,9 @@ def execute_job_in_process(
             asset_selection=frozenset(asset_selection),
         )
 
+    def with_resources(self, resources: Optional[Mapping[str, Any]]) -> "Definitions":
```

Comment:
If adding this, do we need the resources param on `component_defs`? e.g. `component_defs(component).with_resources(...)`

Reply:
Yeah I could go either way.

Reply:
I liked resources appearing in the typeahead so it was obvious.